hamilton/plugins/h_pandera.py

```diff
@@ -0,0 +1,91 @@
+import typing
+from typing import List
+
+import pandera
+from pandera import typing as pa_typing
+
+from hamilton import node
+from hamilton.data_quality import base as dq_base
+from hamilton.function_modifiers import InvalidDecoratorException
+from hamilton.function_modifiers import base as fm_base
+from hamilton.function_modifiers import check_output as base_check_output
+from hamilton.function_modifiers.validation import BaseDataValidationDecorator
+from hamilton.htypes import custom_subclass_check
+
+
+class check_output(BaseDataValidationDecorator):
+    def __init__(
+        self,
+        importance: str = dq_base.DataValidationLevel.WARN.value,
+        target: fm_base.TargetType = None,
+    ):
+        """Specific output-checker for pandera schemas. This decorator utilizes the output type of the function, which has
+        to be of type pandera.typing.pandas.DataFrame or pandera.typing.pandas.Series, with an annotation argument.
+
+        :param schema: The schema to use for validation. If this is not provided, then the output type of the function is used.
+        :param importance: Importance level (either "warn" or "fail") -- see documentation for check_output for more details.
+        :param target: The target of the decorator -- see documentation for check_output for more details.
+
+        Let's look at equivalent examples to demonstrate:
+
+        .. code-block:: python
+            :name: "@check_output using output type"
+
+            import pandera as pa
+            import pandas as pd
+            from hamilton.plugins import h_pandera
+            from pandera.typing.pandas import DataFrame
+
+            class MySchema(pa.DataFrameModel):
+                a: int
+                b: float
+                c: str = pa.Field(nullable=True)  # For example, allow None values
+                d: float    # US dollars
+
+            @h_pandera.check_output_schema()
+            def foo() -> DataFrame[MySchema]:
+                return pd.DataFrame() # will fail
+
+        .. code-block:: python
+            :name: "@check_output with passed in type"
+
+            from hamilton import function_modifiers
+
+            @function_modifiers.check_output(schema=MySchema)
```

Comment:
```suggestion
            from hamilton.function_modifiers import check_output

            @check_output(schema=MySchema)
```

Reply:
Want to make it clear where its coming from as they're named the same